src/main/java/com/microsoft/jenkins/kubernetes/wrapper/ResourceUpdaterMap.java

```diff
@@ -7,42 +7,60 @@
 import io.kubernetes.client.models.V1Job;
 import io.kubernetes.client.models.V1Namespace;
 import io.kubernetes.client.models.V1Pod;
-import io.kubernetes.client.models.V1ReplicaSet;
 import io.kubernetes.client.models.V1ReplicationController;
 import io.kubernetes.client.models.V1Secret;
 import io.kubernetes.client.models.V1Service;
 import io.kubernetes.client.models.V1beta1DaemonSet;
 import io.kubernetes.client.models.V1beta1Ingress;
 import io.kubernetes.client.models.V1beta1ReplicaSet;
+import org.apache.commons.lang3.tuple.Pair;
 
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.Map;
 
-public final class ResourceUpdaterMap extends HashMap<Class<?>, Class<? extends ResourceManager.ResourceUpdater>> {
-    private static final Map<Class<?>, Class<? extends ResourceManager.ResourceUpdater>> INSTANCE =
+public final class ResourceUpdaterMap extends HashMap<Class<?>,
+        Pair<Class<? extends ResourceManager>, Class<? extends ResourceManager.ResourceUpdater>>> {
```

Comment:
`Map<Class<?>, Class<? extends ResourceManager.ResourceUpdater>>` =>
`Map<Class<?>,Pair<Class<? extends ResourceManager>, Class<? extends ResourceManager.ResourceUpdater>>>`
If you want to construct a ResourceUpdater, you need to construct the ResourceManager first.